torch/_dynamo/metrics_context.py

```diff
@@ -67,14 +67,14 @@ def increment(self, metric: str, value: int) -> None:
             self._metrics[metric] = 0
         self._metrics[metric] += value
 
-    def set(self, metric: str, value: Any) -> None:
+    def set(self, metric: str, value: Any, overwrite: bool = False) -> None:
```

Comment:
@ezyang you didn't like this when I added it before, did you? Unlike the remote cache case though I don't really know if there's an alternative here.

Reply:
I actually don't remember the argument against `overwrite`. I guess in general we wouldn't expect individual values in the metrics context to change within the same context. If it's valid for them to change, then maybe we'd be losing some detail by overriding with the last value. I don't have the context for these new fields. Why is it expected they might change and we'd only want to log the last-seen value?

Reply:
Yeah, I didn't like it. Is there an aggregate we can compute instead? Even a list append?